src/contexts/navigator.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) Garuda Labs, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import * as Errors from 'hyperview/src/services/navigator/errors';
+import React, { createContext, useContext } from 'react';
+import { Element } from 'hyperview/src/services/navigator/types-legacy';
+
+type Props = { children: React.ReactNode };
+
+export type NavigatorCache = {
+  routeMap?: Map<string, string>;
+  elementMap?: Map<string, Element>;
+  initialRouteName?: string;
+};
+
+/**
+ * Context used to store runtime information about the navigator and urls
+ * Each navigator creates its own context
+ * Urls defined in <route> elements are stored in the routeMap by their key
+ */
+const NavigatorMapContext = createContext<NavigatorCache | null>(null);
+
+function useMapContext() {
+  const context = useContext(NavigatorMapContext);
+  if (context === undefined) {
+    throw new Errors.HvNavigatorMapError(
+      'useMapContext must be used within a NavigatorMapContext',
+    );
+  }
+  return context;
+}
+
+/**
+ * Cache the initial route name for the <navigator>
+ */
+function SetInitialRouteName(name: string) {
```

Comment:
by convention, function names are `camelCase` - we reserve capitalization for the React functional components (also apply to other functions in this file)
```suggestion
function setInitialRouteName(name: string) {
```